<launch>

  <!-- Generate/Load robot description file -->
  <include file="$(find usb_cam)/launch/usb_cam-test.launch" />


  <!-- Display faces on the robot's head -->
  <node pkg="baxter_openpucp" name="faces"  type="faces.py" />

  <!-- Moves head and the eyes blink -->
  <node pkg="baxter_openpucp" name="autonomos_life"  type="autonomos_life.py" />

  <!-- Moves head and the eyes blink -->
  <node pkg="baxter_openpucp" name="hugh_vision"  type="hugh_vision.py" output="screen" />


  <arg name="use_opencv3" default="false" />
  <arg name="use_opencv3_1" default="false" />
  <arg name="use_opencv3_2" default="false" />
  <arg name="use_opencv3_3" default="$(arg use_opencv3)" />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="false" doc="Specify whether the node displays a window to show edge image" />

  <arg unless="$(arg use_opencv3)"
       name="face_cascade_name" default="/opt/ros/kinetic/share/OpenCV-3.3.1-dev/haarcascades/haarcascade_frontalface_alt.xml" doc="Face dtection cascade Filename" />

  <arg unless="$(arg use_opencv3)"
       name="eyes_cascade_name" default="/opt/ros/kinetic/share/OpenCV-3.3.1-dev/haarcascades/haarcascade_eye_tree_eyeglasses.xml" doc="Eyes dtection cascade Filename" />

  <node pkg="opencv_apps" type="face_detection" name="face_detection">
    <remap from="image" to="/usb_cam/image_raw" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="face_cascade_name" value="$(arg face_cascade_name)" />
    <param name="eyes_cascade_name" value="$(arg eyes_cascade_name)" />
  </node>

  <node pkg="baxter_openpucp" name="seg_cam" type="seg_cam.py" output="screen" />

</launch>
